packages/runtime-core/src/componentPublicInstance.ts

```diff
@@ -207,8 +207,9 @@ type PublicPropertiesMap = Record<string, (i: ComponentInternalInstance) => any>
  */
 const getPublicInstance = (
   i: ComponentInternalInstance | null
-): ComponentPublicInstance | null =>
-  i && (i.proxy ? i.proxy : getPublicInstance(i.parent))
+): ComponentPublicInstance | ComponentInternalInstance['exposed'] | null =>
+  i &&
+  (i.proxy ? (i.exposed ? i.exposed : i.proxy) : getPublicInstance(i.parent))
```

Comment:
A bit nitpicking here, but I think this function becomes quite complicate and a bit hard to read. Maybe it's better to convert to a named function?